src/middleware/ipFilter.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+module.exports = setup;
+
+const _ = require('lodash');
+const iputil = require('ip');
+const rangeCheck = require('range_check');
+const createError = require('http-errors');
+
+ /**
+  * testExplicitIp Test an Explicit IP against the list of constraints
+  * @param  {String} ip          IP address of caller
+  * @param  {String} constraints IPs to restrict or allow
+  * @param  {String} mode        mode of operation deny/allow
+  * @return {Boolean}             True or False, if we should block or allow and matched
+  */
+function testExplicitIp(ip, constraints, mode) {
```

Comment:
since those functions are internal (`testExplicitIp`, `testCidrBlock`) I would use a boolean instead of mode for quicker comparison: `isAllow`.
this way we'll have the smaller footprint possible